### ISSUE TITLE
On macOS `Process.PrivateMemorySize64` returns `0`

### ISSUE THREAD

**Issue by nwoolls** [dotnet/runtime#105665]
### Description

Accessing `Process.PrivateMemorySize64` on macOS returns `0`. This was initially reported, and fixed, for Linux [here](https://github.com/dotnet/runtime/issues/23284). Follow-up comments mention macOS but the issue was closed.

The implementation of `AssertNonZeroAllZeroDarwin` seems to indicate this is acknowledged, if not by-design. But the XML comments for `Process.PrivateMemorySize64` doesn't mention any such limitation.

Not sure if this is a bug or feature request, but I'd love to see this implemented for macOS. We're trying to troubleshoot some memory usage but coming up short in our dev environments because of this.

### Reproduction Steps

```bash
mkdir app-macos-PrivateMemorySize64
cd app-macos-PrivateMemorySize64
dotnet new console
```

```csharp
using System.Diagnostics;

var process = Process.GetCurrentProcess();
Console.WriteLine(process.PrivateMemorySize64);
```

```bash
dotnet run
```

### Expected behavior

Outputs "the amount of private memory, in bytes, allocated for the associated process".

### Actual behavior

Outputs `0`.

### Regression?

No

### Known Workarounds

None

### Configuration

```bash
dotnet --version
8.0.303
```

```bash
sw_vers
ProductName:		macOS
ProductVersion:		14.5
BuildVersion:		23F79
```

```bash
uname -a
Darwin nathanialsmbpm2.lan 23.5.0 Darwin Kernel Version 23.5.0: Wed May  1 20:14:38 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6020 arm64
```

**Comment by ShawnWu33** [comment]
Hi, did a bit investigation on this. Apology first if the terminologies are not accurate enough as I am new to this community.
## TL;DR
The private memory size of a process is not exposed in macOS (OSX) SDK, hence the value remains default int value as 0;

## Investigation details
1. The `Process.PrivateMemorySize64` having a getter function which will first ensure process information being retrieved. ([code](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs#L498))
2. When process information did not existed, it trigger ProcessManager to retrieve the process information ([code](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs#L984))
3. The process manager invoke `Interop.libproc.GetProcessInfoById(pid)` to get process information ([code](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.OSX.cs#L44)), which eventually hitting external macOS API proc_pidinfo ([code](https://github.com/dotnet/runtime/blob/main/src/libraries/Common/src/Interop/OSX/Interop.libproc.GetProcessInfoById.cs#L116))
4. Per (macOS header file)[https://opensource.apple.com/source/xnu/xnu-1504.9.17/bsd/sys/proc_info.h.auto.html] (screenshot attached), the response from that API incldues only virtual memory size and shared memory size. Hence the field `PrivateBytes` in ProcessInfo is not assigned, remains default value.
<img width="500" alt="image" src="https://github.com/user-attachments/assets/cf049e18-a22c-498f-869b-9b4667451975">

5. At the end returning field `PrivateBytes` as 0, from _processInfo ([code](https://github.com/dotnet/runtime/blob/112ef3d680c50ae3c64d7c129b658d9cd00a5a3d/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs#L499)).

## Postscript
This issue is known from very first commit of ProcessManager.OSX, It is a [comment](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.OSX.cs#L70) there saying:
```
// Set the values we have; all the other values don't have meaning or don't exist on OSX
```
So I don't see a proper fix available at curren time point.

**Comment by nwoolls** [comment]
@adamsitnik I've staged a PR that addresses this shortcoming on macOS.
